x-pack/test/functional/services/machine_learning/job_source_selection.ts

```diff
@@ -26,5 +26,11 @@ export function MachineLearningJobSourceSelectionProvider({ getService }: FtrPro
       await testSubjects.clickWhenNotDisabled(`savedObjectTitle${sourceName}`);
       await testSubjects.existOrFail('mlPageJobTypeSelection');
     },
+
+    async selectSourceForIndexBasedDataVisualizer(sourceName: string) {
+      await this.filterSourceSelection(sourceName);
+      await testSubjects.clickWhenNotDisabled(`savedObjectTitle${sourceName}`);
+      await testSubjects.existOrFail('mlPageDataVisualizer');
```

Comment:
If the loading takes more than 2.5 seconds here, that step will fail. If we have bigger data or slower machines I can see that happen, so I would suggest to give it a slightly higher timeout (maybe 10 seconds) like this:
```
await testSubjects.existOrFail('mlPageDataVisualizerIndexData', { timeout: 10 * 1000 });
```

Reply:
Done in https://github.com/elastic/kibana/pull/51832/commits/dfde35a493e596fe1db4d22b8d6dc00e0d14de52